Makefile

```diff
@@ -98,66 +108,32 @@ endif
 # Default target is to build the supervisor image
 all: supervisor
 
-# Settings to make the run-supervisor target work behind a proxy
-DOCKERD_PROXY=tools/dind/config/services/docker.service.d/proxy.conf
-${DOCKERD_PROXY}:
-	rm -f ${DOCKERD_PROXY}
-	if [ -n "${rt_http_proxy}" ]; then \
-		proxies="\"HTTP_PROXY=${rt_http_proxy}\""; \
-		proxies="$${proxies[*]} \"http_proxy=${rt_http_proxy}\""; \
-	fi; \
-	if [ -n "${rt_https_proxy}" ]; then \
-		proxies="$${proxies[*]} \"HTTPS_PROXY=${rt_https_proxy}\""; \
-		proxies="$${proxies[*]} \"https_proxy=${rt_https_proxy}\""; \
-	fi; \
-	if [ -n "${rt_no_proxy}" ]; then \
-		proxies="$${proxies[*]} \"no_proxy=${rt_no_proxy}\""; \
-	fi; \
-	if [ -n "${proxies}" ]; then \
-		echo "[Service]" > ${DOCKERD_PROXY}; \
-		echo "Environment=$${proxies[*]}" >> ${DOCKERD_PROXY}; \
-	else \
-		touch ${DOCKERD_PROXY}; \
-	fi
-
 supervisor-tar:
-	docker save --output tools/dind/supervisor-image.tar $(SUPERVISOR_IMAGE)
+	cd tools/dind \
+	&& mkdir -p config \
+	&& docker save --output config/supervisor-image.tar $(SUPERVISOR_IMAGE)
 
-supervisor-dind: ${DOCKERD_PROXY} supervisor-tar
+supervisor-conf:
 	cd tools/dind \
-	&& docker build \
-		$(DOCKER_HTTP_PROXY) \
-		$(DOCKER_HTTPS_PROXY) \
-		$(DOCKER_NO_PROXY) \
-		${DOCKER_BUILD_OPTIONS} \
-		--no-cache=$(DISABLE_CACHE) \
-		--build-arg PASSWORDLESS_DROPBEAR=$(PASSWORDLESS_DROPBEAR) \
-		-t $(IMAGE) .
+	&& mkdir -p config \
+	&& echo "SUPERVISOR_IMAGE=$(call repo,$(SUPERVISOR_IMAGE))" > config/supervisor.conf \
+	&& echo "SUPERVISOR_TAG=$(call tag,$(SUPERVISOR_IMAGE))" >> config/supervisor.conf \
+	&& echo "LED_FILE=/dev/null" >> config/supervisor.conf
+
+supervisor-dind: supervisor-tar supervisor-conf
 
-run-supervisor: stop-supervisor supervisor-dind
+run-supervisor: supervisor-dind
 	cd tools/dind \
-	&& echo "SUPERVISOR_IMAGE=$(SUPERVISOR_IMAGE)" > config/localenv \
-	&& echo "PRELOADED_IMAGE=$(PRELOADED_IMAGE)" >> config/localenv \
-	&& echo "SUPERVISOR_EXTRA_MOUNTS=$(SUPERVISOR_EXTRA_MOUNTS)" >> config/localenv; \
-	if [ -n "$(rt_http_proxy)" ]; then \
-		echo "HTTP_PROXY=$(rt_http_proxy)" >> config/localenv \
-		&& echo "http_proxy=$(rt_http_proxy)" >> config/localenv; \
-	fi; \
-	if [ -n "$(rt_https_proxy)" ]; then \
-		echo "HTTPS_PROXY=$(rt_https_proxy)" >> config/localenv \
-		&& echo "https_proxy=$(rt_https_proxy)" >> config/localenv; \
-	fi; \
-	if [ -n "$(rt_no_proxy)" ]; then \
-		echo "no_proxy=$(rt_no_proxy)" >> config/localenv; \
-	fi \
-	&& docker run -d --name $(CONTAINER_NAME) --privileged ${SUPERVISOR_DIND_MOUNTS} $(IMAGE)
+	&& ./resinos-in-container/resinos-in-container.sh \
+		--detach \
+		--config "$$(pwd)/$(CONFIG_FILENAME)" \
+		--image resin/resinos:2.12.5_rev1-intel-nuc \
```

Comment:
Should that be configurable via a script argument? Or at least moved to a file-level constant so it's discoverable and easily updated?

Reply:
Good point, done